tests/core/pyspec/eth2spec/test/merge/unittests/test_terminal_validity.py

```diff
@@ -0,0 +1,150 @@
+from eth2spec.test.exceptions import BlockNotFoundException
+from eth2spec.utils.ssz.ssz_typing import uint256
+from eth2spec.test.helpers.fork_choice import (
+    prepare_empty_pow_block,
+)
+from eth2spec.test.context import spec_state_test, with_merge_and_later
+
+
+# Copy of conditional merge part of `on_block(store: Store, signed_block: SignedBeaconBlock)` handler
+def validate_transition_execution_payload(spec, execution_payload):
+    pow_block = spec.get_pow_block(execution_payload.parent_hash)
+    pow_parent = spec.get_pow_block(pow_block.parent_hash)
+    assert spec.is_valid_terminal_pow_block(pow_block, pow_parent)
+
+
+def run_process_merge_execution_payload(spec, block, parent_block, payload,
```

Comment:
This doesn't run `process_merge_execution_payload` (which isn't a function I believe) nor `process_execution_payload`. Instead, it runs `is_valid_terminal_pow_block`.

I'm also unsure why we are yielding anything here. Is this a new vector type? And if it is, shouldn't a result -- True or False be provided?

It seems that this function is incorrectly named and that it *should not* be yielding any test vectors